release-notes/7.0/preview/api-diff/preview1/Microsoft.AspNetCore.App/7.0-preview1_Microsoft.Extensions.Configuration.md

```diff
@@ -0,0 +1,159 @@
+# Microsoft.Extensions.Configuration
+
+``` diff
+ namespace Microsoft.Extensions.Configuration {
+     public class ChainedConfigurationProvider : IConfigurationProvider, IDisposable {
+-        public IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string parentPath);
++        public IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string? parentPath);
+-        public void Set(string key, string value);
++        public void Set(string key, string? value);
+-        public bool TryGet(string key, out string value);
++        public bool TryGet(string key, out string? value);
+     }
+     public static class CommandLineConfigurationExtensions {
+-        public static IConfigurationBuilder AddCommandLine(this IConfigurationBuilder builder, Action<CommandLineConfigurationSource> configureSource);
++        public static IConfigurationBuilder AddCommandLine(this IConfigurationBuilder builder, Action<CommandLineConfigurationSource>? configureSource);
+-        public static IConfigurationBuilder AddCommandLine(this IConfigurationBuilder configurationBuilder, string[] args, IDictionary<string, string> switchMappings);
++        public static IConfigurationBuilder AddCommandLine(this IConfigurationBuilder configurationBuilder, string[] args, IDictionary<string, string>? switchMappings);
```

Comment:
@eerhardt, why do some of the string/string dictionaries and enumerables use `string, string` and some use `string, string?`?  Do they actually differ in semantics as to whether the values are expected to be null?